src/placementEngine.test.js

```diff
@@ -1,33 +1,100 @@
 import mockStorage from './mockStorage'
 import placementEngine from './placementEngine'
+import stateMachine from './stateMachine'
+import { getUser } from './userState'
+
+jest.mock('./stateMachine', () => jest.fn())
+jest.mock('./userState', () => ({ getUser: jest.fn() }))
+
+// Mock the now function.
+Date.now = jest.fn(() => 123)
 
 describe('placementEngine', () => {
-  const userAgent = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.131 Safari/537.36'
-  const a = { promoId: 1 }
-  const b = { promoId: 2, constraints: 'user.visits > 1' }
-  const c = { promoId: 3, constraints: 'browser.name = "Chrome"' }
-  const d = { promoId: 4, groupId: 1, constraints: 'groupId NOT IN user.blocked.groups' }
-  const e = { promoId: 5, constraints: 'custom.foo = "bar"' }
-  const promos = [a, b, c, d, e]
-  const custom = { foo: 'bar' }
-
-  // Stub the user agent.
-  Object.defineProperty(window.navigator, 'userAgent', { value: userAgent })
-
-  it('increments the number of visits', () => {
```

Comment:
:clap: on removing constraint checking tests from the `placementEngine.test.js`.